MonstieWash/Assets/Scripts/PlayerHand.cs

```diff
@@ -4,49 +4,42 @@
 
 public class Playerhand : MonoBehaviour
 {
+    //Public
+    public Vector3 handPosition { get; private set; }
+
     //Unity Inspector
     [SerializeField]
     [Range(1.0f, 50.0f)]
     private float cursorSpeed = 20f;
 
     //Private
-    private Transform m_transform;
-    private Vector3 m_handPosition;
     private float m_moveHorizontal;
     private float m_moveVertical;
-	
-    // Start is called before the first frame update
-    void Start()
-    {
-        m_transform = gameObject.transform;
-    }
 
     // Update is called once per frame
-    void Update()
+    private void Update()
     {
-        m_handPosition = Camera.main.WorldToScreenPoint(m_transform.position);
         m_moveHorizontal = Input.GetAxis("Horizontal");
         m_moveVertical = Input.GetAxis("Vertical");
 
-        //Boundary Check X-axis
-        if(m_handPosition.x > Screen.width && m_moveHorizontal > 0)
-        {
-            m_moveHorizontal = 0;
-        }
-        else if(m_handPosition.x < 0 && m_moveHorizontal < 0)
-        {
-            m_moveHorizontal = 0;
-        }
-        //Boundary Check Y-axis
-        if(m_handPosition.y > Screen.height && m_moveVertical > 0)
-        {
-            m_moveVertical = 0;
-        }
-        else if(m_handPosition.y < 0 && m_moveVertical < 0)
-        {
-            m_moveVertical = 0;
-        }
-        
-        m_transform.position = transform.position + new Vector3(m_moveHorizontal * cursorSpeed * Time.deltaTime, m_moveVertical * cursorSpeed * Time.deltaTime, 0);
+        transform.position = CalculateMovement();
+        handPosition = Camera.main.WorldToScreenPoint(transform.position);
```

Comment:
This one is my bad, handPosition should probably be assigned in world point because it will need to be compared to many other objects in scene which also default to world point. I only changed it to screen point because I had to compare to screen edges

Reply:
Ah I see, well in that case hand position can be found just from transform.position, so would we want to still keep handPosition at all? I use it currently in the eraser code but I don't necessarily need it.

Reply:
i'd remove it since its not needed